nemo/src/model/rule_model/term.rs

```diff
@@ -367,6 +375,58 @@ impl Term {
             Term::Function(sub) => sub.subterms.iter().any(Self::aggregate_subterm_recursive),
         }
     }
+
+    /// Evaluates a constant (numeric) term
+    /// We do this by casting everything to `f64` as it seems like the most general number type.
+    /// Returns `None` if this not possible.
+    pub fn evaluate_constant_numeric(&self) -> Option<Double> {
+        match self {
+            Term::Primitive(primitive) => match primitive {
+                PrimitiveTerm::Constant(constant) => {
+                    if let DataValueT::Double(value) = PrimitiveType::Float64
+                        .ground_term_to_data_value_t(constant.clone())
+                        .ok()?
+                    {
+                        Some(value)
+                    } else {
+                        None
+                    }
+                }
+                PrimitiveTerm::Variable(_) => todo!(),
+            },
+            Term::Binary(binary) => {
+                let (left, right) = binary.terms();
+                let value_left = left.evaluate_constant_numeric()?;
+                let value_right = right.evaluate_constant_numeric()?;
+
+                match binary {
+                    BinaryOperation::Addition(_, _) => Some(value_left + value_right),
+                    BinaryOperation::Subtraction(_, _) => Some(value_left - value_right),
+                    BinaryOperation::Multiplication(_, _) => Some(value_left * value_right),
+                    BinaryOperation::Division(_, _) => Some(value_left / value_right),
```

Comment:
if `value_left = value_right = 0`, this will produce a `Double` holding a `NaN`, which shouldn't happen, and since we currently only have checks for `NaN` in the `Double` constructors, this will just lead to weird behaviour later on.